api/util.py

```diff
@@ -88,11 +88,12 @@ def path_from_hash(hash_):
     """
     create a filepath from a hash
     e.g.
-    hash_ = 01b395a1cbc0f218
+    hash_ = v0-sha384-01b395a1cbc0f218
     will return
-    0/1/b/3/9/5/a/1/01b395a1cbc0f218
+    v0/sha384/01/b3/v0-sha384-01b395a1cbc0f218
     """
-    path = [hash_[i] for i in xrange(8)]
+    hash_version, hash_alg, actual_hash = hash_.split('-')
+    path = [hash_version, hash_alg] + [actual_hash[2 * i : 2 * i + 2] for i in range(2)]
```

Comment:
List comprehension is a key language feature of Python and we most-definitely won't stop using it.

I agree that this instance is totally pointless, though, and should be replaced by your first suggestion, but corrected:

```
[actual_hash[0:2], actual_hash[2:4]]
```

One more thing: We should be using tuples instead of arrays it indicate immutability when appropriate.